### PR TITLE
Enable/Disable Discount in POS using POS Profile #11748

### PR DESCRIPTION
**Pull request by jay-parikh**
Added new Checkbox to POS Profile - "Allow user to edit Discount", which makes discount input field show/hide and make discount btn to Numpad disable.
Please follow below gif:
enable disable discount

### CODE REVIEW

File: erpnext/selling/page/point_of_sale/point_of_sale.js [hunk 1]
```diff
@@ -714,6 +715,21 @@ class POSCart {
...
+			if(!this.frm.allow_edit_rate) {
+				return ['Rate'];
+			} else {
+				return ['Disc'];
+			}
+		} else {
+			return [];
+		}
+	}
+
```
Comment (rohitwaghchaure) [inline erpnext/selling/page/point_of_sale/point_of_sale.js]: @jay-parikh 

You can also write above function as below
```
disable_numpad_control() { 
	let disabled_btns = [];

	if(!this.frm.allow_edit_rate) {
		disabled_btns.push('Rate')
	}

	if(!this.frm.allow_edit_discount) {
		disabled_btns.push('Disc')
	}

	return disabled_btns;
}
```

Comment (jay-parikh) [reply erpnext/selling/page/point_of_sale/point_of_sale.js]: @rohitwaghchaure Thanks for feedback. Let me rewrite and check.

**Comment by rohitwaghchaure** [comment]
@jay-parikh Check codacy issue

**Comment by jay-parikh** [comment]
Noted! Thanks

**Review by jay-parikh — COMMENTED** [review]
@rohitwaghchaure Done suggested changes. Please review.

**Review by rohitwaghchaure — APPROVED** [review]

**Comment by olamide2** [comment]
Hello @jay-parikh 

This does not work for offline PoS

Is their any way we can make this work for offline PoS?

Regards

**Pull request merged**